site/app/views/submission/HomeworkView.php

```diff
@@ -1186,14 +1187,41 @@ public function showGradeInquiryDiscussion(GradedGradeable $graded_gradeable, bo
                     $gc_title = $component->getTitle();
                 }
 
+
+                $grade_inquiry_posts_for_id = $grade_inquiry_posts[$grade_inquiry->getId()];
+                $author_user_ids = array_map(function ($post) {
+                    return $post["user_id"];
+                }, $grade_inquiry_posts_for_id);
+                $author_user_groups = $queries->getAuthorUserGroups($author_user_ids);
+
+                $instructor_full_access = [];
+                $limited_access_grader = [];
+
+                foreach ($author_user_groups as $author) {
+                    $limited_access_grader[$author["user_id"]] = $author["user_group"] === User::GROUP_LIMITED_ACCESS_GRADER;
+                    $instructor_full_access[$author["user_id"]] = $author["user_group"] <= User::GROUP_FULL_ACCESS_GRADER;
+                }
+
                 // format posts
                 $posts = [];
                 foreach ($grade_inquiry_posts[$grade_inquiry->getId()] as $post) {
                     if (empty($post)) {
                         break;
                     }
-                    $is_staff = $this->core->getQueries()->isStaffPost($post['user_id']);
-                    $name = $this->core->getQueries()->getUserById($post['user_id'])->getDisplayedGivenName();
+                    $is_staff = $queries->isStaffPost($post['user_id']);
```

Comment:
This is going to result in the same performance issues I commented on above.  This one seems relatively simple to convert to a single larger query.